test/sync/src/commonMain/kotlin/io/realm/test/mongodb/util/AdminApi.kt

```diff
@@ -120,113 +120,103 @@ open class AdminApiImpl internal constructor(
     @Serializable
     data class ServerApp(val client_app_id: String, val _id: String)
 
-    // FIXME Do not rely on this API to create users and use EmailPasswordAuth wrapper is ready
-    //  see https://github.com/realm/realm-kotlin/issues/433
     init {
-        // Must be initialized on same thread as the constructor to allow initializing the lateinit
-        // properties on native
+        // Work around issues on Native with the Ktor client being created and used
+        // on different threads.
         runBlocking(Dispatchers.Unconfined) {
             // Log in using unauthorized client
-            loginResponse =
+            val loginResponse =
                 defaultClient("$appName-unauthorized", debug).typedRequest<LoginResponse>(
                     HttpMethod.Post,
                     "$url/auth/providers/local-userpass/login"
                 ) {
                     contentType(ContentType.Application.Json)
                     body = mapOf("username" to "unique_user@domain.com", "password" to "password")
                 }
-
-            // Setup authorized client for the rest of the requests
+            //
+            // // Setup authorized client for the rest of the requests
+            // // Client is currently being constructured for each network reques to work around
+            // //  https://github.com/realm/realm-kotlin/issues/480
             val accessToken = loginResponse.access_token
-            client = defaultClient("$appName-authorized", debug) {
-                defaultRequest {
-                    headers {
-                        append("Authorization", "Bearer $accessToken")
+            client = {
+                defaultClient("$appName-authorized", debug) {
+                    defaultRequest {
+                        headers {
+                            append("Authorization", "Bearer $accessToken")
+                        }
+                    }
+                    install(JsonFeature) {
+                        serializer = KotlinxSerializer()
+                    }
+                    install(Logging) {
+                        // Set to LogLevel.ALL to debug Admin API requests. All relevant
```

Comment:
isn't this sensitive as a default for production? it could leak some API keys/secrets ...

Reply:
The AdminAPI is only used for our tests, user code never calls it. So I think it should be safe, but after looking at the logs, it does clutter it a bit, so I'll reduce it to something less verbose.